README
2015-09-15 mar 03:50 

Traduzione della storia di terminal-quest

ssh Andreas@192.168.43.23 (o l'ip che è)
in kano
x11vnc

in x220
arf@arf-ThinkPad-X220:~/Didattica/Kano/TERMINAL-QUEST$ ./VNC-Viewer-5.2.3-Linux-x86
passw: pippo1

OK!

###
a caroline clarck avevo scritto:
https://github.com/KanoComputing/terminal-quest/issues/15

As far as I understood I should do the following:

Translate...

1) the file names of the objects in
/usr/share/pyshared/linux-story/ascii_assets/story-files

2) the explanations of commands in those files, such as CAT, for instance

3) the names of those files, wherever they appear in all the appropriate *.py

4) the story texts, in the 92 files:
/usr/share/pyshared/linux-story/story/challenges/challenge_*.py

5) all the appropriate names in
/usr/share/pyshared/linux-story/story/trees/default_trees.yml

###
2015-09-17 gio 05:00 

Bene lavoro con X220 collegato via wirelesse alla PB
RPI collegato alla PB via ethernet

ssh Andreas@192.168.1.X (o l'ip che è)
in kano
x11vnc

in x220
arf@arf-ThinkPad-X220:~/Didattica/Kano/TERMINAL-QUEST$ ./VNC-Viewer-5.2.3-Linux-x86
passw: pippo1

Una volta nel RPI in ~/terminal-quest

giro il software modificato con 

Una volta nel RPI in ~/terminal-quest

giro il software modificato con 
./bin/linux-story-gui -d challenge 1 1 

invece il software residente, inalterato, gira con

linux-story-gui -d challenge 1 1 

Tradotto la storia del challenge 1 livello 1

Ho tradotto i nomi delle cartelle (fake)
my-house
my-room
ma andranno tradotti non solo nei file challenge_*.py

Da fare:

1) Non ho cambiato in 37 e 38 di
   linux-story/story/challenges/challenge_1.py
   ~/my-house/my-room
   in
   ~/la-mia-casa/la-mia-stanza
   Ma questo andrà fatto di concerto in tutti i file interessati 
   grep -r my-house * | wc -l
   rivela 184 file
   grep -r my-room * | wc -l
   69
   L'albero di file esiste e si trova in ~/.linux-story
   ...devo capire come viene generato
   Andrà fatto in modo automatico...

2) deve digerire i caratteri accentati Unicode o ascii iso latin...
   2015-09-17 gio 05:47 FATTO! Aggiungere in testa
   #!/usr/bin/env python
   # coding: utf-8

3) In testa appare la scritta CHALLENGE
   questa si trova in 
   linux_story/gtk3/MenuScreen.pyc
   binario derivato in qualche modo da 
   linux_story/gtk3/Menuscreen.py
   Da capire, chiedere come procedere e correggere...
   By the way: come traduco qui "challenge"?

Per ora ho tradotto

./linux_story/story/challenges/challenge_1.py

Funziona, a parte i fatti precedenti che vanno sistemati per bene.

Per ora ho solo aggiunto il file fra quelli da tracciare

git add linux_story/story/challenges/challenge_1.py

senza committing - devo prima correggere my-house e my-room...?

###
19 settembre

fatto allo stesso modo le variazioni in


./linux_story/story/challenges/challenge_2.py

###
20 settembre

In treno, sull'X220

arf@arf-ThinkPad-X220:~/Didattica/Kano/TERMINAL-QUEST/terminal-quest$ ls -l linux_story/ascii_assets/story_files/ | awk '{print $9}' > FOO





cp alarm it/sveglia
cp alice-and-wonderland it/alice-e-il-paese-delle-meraviglie   # tradurre "COLD"? Fatto ma bisogna vedere se è davvero COLD...
cp apple it/mela
cp banana it/banana
cp bed_farmhouse it/letto_fattoria
cp bed_my-room it/letto_camera-mia
cp bed_parents-room it/letto_camera-genitori
cp bench it/panchina
cp Bernard it/Bernardo
cp bernards-diary-1 it/diario-bernardo-1   # tradurre
cp bernards-diary-2 it/diario-bernardo-2   # tradurre
cp bernards-hat it/cappello-bernardo
cp best-horn-in-the-world-correct.sh it/miglior-clackson-del-mondo-corretto.sh   # tradurre
cp best-horn-in-the-world-incorrect.sh it/miglior-clackson-del-mondo-scorretto.sh   # tradurre
cp best-shed-maker-in-the-world.sh it/miglior-fabbricatore-di-capanni-del-mondo.sh   # tradurre
cp bull it/toro
cp cake it/torta
cp cap it/berretto
cp carrots it/carote
cp CAT it/CAT   # tradurre contenuto
cp CD it/CD   # tradurre contenuto
cp chair it/sedia
cp chick it/pulcino
cp chicken it/pollo
cp Clara it/Clara
cp Cobweb it/Gelsomino   # o altro nome proprio adatto a un cavallo
cp comic-book it/fumetto
cp computer it/computer
cp country-boy it/garzone   # contadinello?
cp croissant it/cornetto
cp Dad it/babbo
cp Daisy it/Violetta   # o altro noime da mucca
cp dame it/signora   # ?
cp desk it/scrivania
cp dog it/cane
cp dress it/vestito
cp ECHO it/ECHO   # tradurre contenuto
cp Edith it/Edith   # o altro nome (vedere storia)
cp Edward it/Edoardo
cp Eleanor it/Eleonora
cp empty-bottle it/bottiglia-vuota
cp farms it/fattorie
cp fence it/recinto
cp fighter it/combattente
cp flowers it/fiori
cp grapes it/grappoli
cp grumpy-man it/brontolone
cp hammer it/martello
cp jumper it/felpa
cp librarian it/bibliotecario
cp little-boy it/ragazzino
cp LS it/LS   # tradurre contenuto
cp map it/mappa
cp Mayor it/sindaco
cp milk it/latte   # tradurre "MILK"
cp MKDIR it/MKDIR   # tradurre contenuto
cp Mum it/mamma
cp mums-diary it/diario-mamma
cp MV it/MV   # tradurre contenuto
cp NANO it/NANO   # tradurre contenuto
cp nerd it/smanettone
cp newspaper it/giornale   # tradurre contenuto 
cp note_greenhouse it/foglietto_serra
cp note_kitchen it/foglietto_cucina
cp note_my-room it/foglietto_camera-mia   # sistemare problema lettera ò
cp note_town it/foglietto_paese
cp onion it/cipolla
cp oven it/stufa
cp photocopier.sh it/fotocopiatrice.sh
cp picture it/quadro   # ?
cp pie it/tortina   # ?
cp pumpkin it/zucca
cp redwall it/pinocchio   # se redwall è effettivamente un racconto per bambini...
cp Ruth it/Romina   # o nome adatto (vedere storia)
cp sandwich it/panino
cp saw it/sega
cp screwdriver it/cacciavite
cp shirt it/camicia
cp skirt it/gonna
cp spanner it/chiave-inglese
cp table it/tavolo
cp tape-measure it/metro
cp tomato it/pomodoro
cp Trotter it/Trogolo   # o altro nome da maiale...
cp trousers it/pantaloni
cp t-shirt it/maglietta
cp tv it/televisione
cp watership-down it/la-collina-dei-conigli
cp wimpy-guy it/tipo-imbranato   # o deboluccio...?
cp window it/finestra
cp witch it/strega
cp young-girl it/ragazzina

Le traduzioni precedenti si riferiscono agli oggetti presenti
sotto forma di ascii artwork in
./linux_story/ascii_assets/story_files/ 

ma ci sono altri termini, che compaiono tutti (credo) in
linux_story/story/trees/default_trees.yaml
che elenco qui sotto

my-room camera-mia
my-house casa-mia
shelves scaffali
parents-room camera-genitori
garden giardino
greenhouse serra
.note .foglietto
.hidden-shelter .riparo-nascosto
town paese
basket cestino
kitchen cucina
east est
shed-shop negozio-capanni   # verificare 
library biblioteca   # verificare
restaurant ristorante
cellar cantina
.chest .scrigno   # verificare
.tiny-chest .piccolo-scrigno   # verificare
.safe .cassaforte   # verificare
farm fattoria
barn fienile   # verificare
shelter riparo   # verificare
farmhouse casa-della-fattoria
toolshed capanno-degli-attrezzi
basement seminterrato   # verificare
public-section sezione-pubblica
Doggy Pippo




messo nello script transl-cp-file.sh che deve girare in
./linux_story/ascii_assets/story_files/ e sempre qui ci deve essere la
directory it, dove vengono copiati file in versione italiana - alcuni da
tradurre anche dentro

###
2015-09-21 Mon 04:47 PM

Portato tutto nel Kano e introdotte le modifiche relative alla creazione
di nuovi file con nomi in italiano in un nuovo branch id nome "file-italiani"

###
2015-09-22 Tue 09:50 PM

Tradotto quasi tutto l'albero

./linux_story/story/trees/default_trees.yaml

Per ogni challenge devo fare le sequenti cose:

- :%s/my-room/camera-mia/
- :%s/my-house/casa-mia/
- tradurre story (lascio per ora la copia inglese commentata)
- tradurre commands (lascio per ora la copia inglese commentata)
- tradurre hints (lascio per ora la copia inglese commentata)

Attenzione, in 

./linux_story/story/trees/default_trees.yaml

ho tradotto anche .note in .foglietto...

Completato i primi 3 challenge.



debian/changelog
linux-story (4.0.0-0) unstable; urgency=low

  * Fixed Gtk3 styling and import issues for packages in Stretch
  * Updated debian dependencies for the project

 -- Team Kano <dev@kano.me>  Fri, 20 Apr 2018 19:40:00 +0100

linux-story (3.15.0-0) unstable; urgency=low

  * Fixed app icon on the Desktop to be the same as on Dashboard

 -- Team Kano <dev@kano.me>  Wed, 20 Dec 2017 14:24:00 +0100

linux-story (3.14.0-0) unstable; urgency=low

  * Applied changes from v3.9.1 es_AR backports to latest
  * Added Loading animation splash window

 -- Team Kano <dev@kano.me>  Mon, 16 Nov 2017 17:04:00 +0100

linux-story (3.9.1-0) unstable; urgency=low

  * i18n: Add updated es_AR translations and fixes (backport)
  * Fix app shortcut to acquire interactive shell environment (backport)

 -- Team Kano <dev@kano.me>  Fri, 21 Jul 2017 13:48:50 +0001

linux-story (1.2-1) unstable; urgency=low

  * Initial release.

 -- Team Kano <dev@kano.me>  Wed, 22 Oct 2014 09:57:49 +0001


linux_story/titles.py
# titles.py
#
# Copyright (C) 2014-2016 Kano Computing Ltd.
# License: http://www.gnu.org/licenses/gpl-2.0.txt GNU GPL v2
#

chapters = {
    1: {
        'start_challenge': 1,
        'end_challenge': 9,
        'title': _('Start exploring')
    },
    2: {
        'start_challenge': 10,
        'end_challenge': 16,
        'title': _('Save a family')
    },
    3: {
        'start_challenge': 17,
        'end_challenge': 22,
        'title': _('Go to the farm')
    },
    4: {
        'start_challenge': 23,
        'end_challenge': 31,
        'title': _('Save Eleanor')
    },
    5: {
        'start_challenge': 32,
        'end_challenge': 39,
        'title': _('Meet the swordmaster')
    },
    6: {
        'start_challenge': 40,
        'end_challenge': 46,
        'title': _('Fight the kidnapper')
    }
}


# Contains the text describing the challenges
challenges = {
    1: {
        'title': _('Wake up!'),
        'chapter': 1
    },
    2: {
        'title': _('Look in your wardrobe'),
        'chapter': 1
    },
    3: {
        'title': _('Look on your shelves'),
        'chapter': 1
    },
    4: {
        'title': _('Find Mum'),
        'chapter': 1
    },
    5: {
        'title': _('Where\'s Dad?'),
        'chapter': 1
    },
    6: {
        'title': _('Visit the town'),
        'chapter': 1
    },
    7: {
        'title': _('Town meeting'),
        'chapter': 1
    },
    8: {
        'title': _('The bell strikes'),
        'chapter': 1
    },
    9: {
        'title': _('Where\'s Mum?'),
        'chapter': 1
    },
    10: {
        'title': _('See more clearly'),
        'chapter': 2
    },
    11: {
        'title': _('Save the girl'),
        'chapter': 2
    },
    12: {
        'title': _('Save the dog'),
        'chapter': 2
    },
    13: {
        'title': _('Food hunt'),
        'chapter': 2
    },
    14: {
        'title': _('Folderton Hero'),
        'chapter': 2
    },
    15: {
        'title': _('Have a closer look'),
        'chapter': 2
    },
    16: {
        'title': _('A gift'),
        'chapter': 2
    },
    17: {
        'title': _('Find your voice'),
        'chapter': 3
    },
    18: {
        'title': _('Visit the farm'),
        'chapter': 3
    },
    19: {
        'title': _('Meet Ruth'),
        'chapter': 3
    },
    20: {
        'title': _('Learn to build'),
        'chapter': 3
    },
    21: {
        'title': _('Hide Ruth and her animals'),
        'chapter': 3
    },
    22: {
        'title': _('Did you hear that?'),
        'chapter': 3
    },
    23: {
        'title': _('Hello Eleanor'),
        'chapter': 4
    },
    24: {
        'title': _('Go east'),
        'chapter': 4
    },
    25: {
        'title': _('Meet Bernard'),
        'chapter': 4
    },
    26: {
        'title': _('Go into the library'),
        'chapter': 4
    },
    27: {
        'title': _('Help Bernard'),
        'chapter': 4
    },
    28: {
        'title': _('Find the librarian'),
        'chapter': 4
    },
    29: {
        'title': _('Talk to Clara'),
        'chapter': 4
    },
    30: {
        'title': _('Did you hear that?'),
        'chapter': 4
    },
    31: {
        'title': _('Explore the shed-shop'),
        'chapter': 4
    },
    32: {
        'title': _('Follow Ruth\'s hint'),
        'chapter': 5
    },
    33: {
        'title': _('Another locked door'),
        'chapter': 5
    },
    34: {
        'title': _('Cave of mysteries'),
        'chapter': 5
    },
    35: {
        'title': _('Set the bird free'),
        'chapter': 5
    },
    36: {
        'title': _('Set off the fireworks'),
        'chapter': 5
    },
    37: {
        'title': _('Answer the riddle'),
        'chapter': 5
    },
    38: {
        'title': _('Go back to the swordmaster\'s.'),
        'chapter': 6
    },
    39: {
        'title': _('Meet the Swordmaster'),
        'chapter': 6
    },
    40: {
        'title': _('Trail of notes'),
        'chapter': 6
    },
    41: {
        'title': _('Meet the note giver'),
        'chapter': 6
    },
    42: {
        'title': _('Unlock the library'),
        'chapter': 6
    },
    43: {
        'title': _('A helping hand'),
        'chapter': 7
    },
    44: {
        'title': _('Go into the rabbithole'),
        'chapter': 7
    },
    45: {
        'title': _('Save everyone'),
        'chapter': 7
    },
    46: {
        'title': _('Break the curse'),
        'chapter': 7
    }
}
